File: src/graph/adapter/fixtures/mixed-repo/sample.go
package sample

import (
  "fmt"
  "net/http"
)

type Worker struct{}

type Runner interface {
  Run() string
}

func Build() *Worker {
  return &Worker{}
}

func (w Worker) Run() string {
  return fmt.Sprintf("%T", http.MethodGet)
}

File: src/graph/adapter/fixtures/mixed-repo/sample.rs
use std::fmt;

pub struct Worker;

pub enum Mode {
    DryRun,
}

pub trait Runner {
    fn run(&self) -> String;
}

pub fn build() -> Worker {
    Worker
}

impl Worker {
    pub fn run(&self) -> String {
        format!("{:?}", fmt::Error)
    }
}
